src/descriptor/mod.rs

```diff
@@ -457,15 +461,14 @@ impl<Pk, Pkh> expression::FromTree for Descriptor<Pk, Pkh> where
     }
 }
 
-impl<Pk, Pkh> FromStr for Descriptor<Pk, Pkh> where
-    Pk: Clone + fmt::Debug + fmt::Display + str::FromStr,
-    Pkh: Clone + fmt::Debug + fmt::Display + str::FromStr,
+impl<Pk> FromStr for Descriptor<Pk> where
+    Pk: MiniscriptKey,
     <Pk as FromStr>::Err: ToString,
-    <Pkh as FromStr>::Err: ToString,
+    <<Pk as MiniscriptKey>::Hash as str::FromStr>::Err: ToString,
```

Comment:
I think this bound is unnecessary because `Pk: MiniscriptKey` already implies `Pk::Hash: str::FromStr`.

Reply:
Oh, no, I see, it's a bound on `Pk::Hash::Err`. Ok.

Reply:
the compiler complains if I remove that